### PR TITLE
feat: add token id in collectible dialog

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #

### CODE REVIEW

**Comment by wenluomask** [comment]
Please make the button unclickable when contract address is correct but id is incorrect.

**Review by guanbinrui — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx [hunk 1]
```diff
@@ -1,48 +1,68 @@
...
-            //         ...tokenDetailed.value,
-            //         asset: assetDetailed.value,
-            //     }),
-            //     PluginServices.Wallet.trustERC721Token(wallet.address, tokenDetailed.value),
-            // ])
-        },
-        deps: [wallet],
-        onSuccess: onClose,
-    })
+    const onSubmit = async () => {
```
Comment (zhouhanseng) [inline packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx:38]: Wrap with `useCallback(async () => { // ...}, [// ...])`

File: packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx [hunk 2]
```diff
@@ -1,48 +1,68 @@
...
+        if (contractDetailLoading || !wallet) return
+
+        const tokenInDB = await PluginServices.Wallet.getERC721Token(address, tokenId)
+        if (tokenInDB) throw new Error(FormErrorType.Added)
+
+        const _tokenDetailed = await erc721TokenDetailedCallback()
```
Comment (zhouhanseng) [inline packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx:44]: change `_tokenDetailed` to `tokenDetailed`

File: packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx [hunk 3]
```diff
@@ -1,48 +1,68 @@
...
+
+        if (
+            (_tokenDetailed && !isSameAddress(_tokenDetailed.info.owner, wallet.address)) ||
+            !_tokenDetailed ||
+            !_tokenDetailed.info.owner
+        ) {
+            throw new Error(FormErrorType.NotExist)
```
Comment (zhouhanseng) [inline packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx:51]: Is there a UX alarm?

File: packages/theme/src/Components/TextField/index.tsx [hunk 4]
```diff
@@ -8,21 +8,25 @@ import {
...
-        // marginBottom: theme.spacing(4),
-    },
     label: {
         fontSize: 12,
         lineHeight: '16px',
         fontWeight: 'bolder',
     },
+    required: {
+        fontSize: 12,
+        lineHeight: '16px',
```
Comment (zhouhanseng) [inline packages/theme/src/Components/TextField/index.tsx:21]: `lineHeight: 16`

File: packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx (hunk 3 above)
Comment (NikoManiac) [reply packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx:51]: The react-hook-form will catch the error and I will show the error message on UI.
```
handleSubmit(onSubmit)(event).catch((error) => {
                setError('tokenId', {
                    type: 'value',
                    message:
                        error.message === FormErrorType.Added
                            ? t.wallets_collectible_been_added()
                            : t.wallets_collectible_error_not_exist(),
                })
            })
```

File: packages/theme/src/Components/TextField/index.tsx (hunk 4 above)
Comment (NikoManiac) [reply packages/theme/src/Components/TextField/index.tsx:21]: > The used value is this unitless <number> multiplied by the element's own font size. [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height)

> `lineHeight: 16`

**Review by zhouhanseng — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**